cluster_management/src/main/java/com/pinterest/rocksplicator/LeaderFollowerStateModelFactory.java

```diff
@@ -266,10 +281,32 @@ public void onBecomeLeaderFromFollower(Message message, NotificationContext cont
           }
         }
 
+        LOG.error("prem: local.seq:" + localSeq + " replica:[" + numActiveReplicas + "/"+ totalReplicas + "]");
+        // 3-Node failure case
+        if (localSeq <= 0 || numActiveReplicas < totalReplicas/2) {
```

Comment:
Can you add a comment on why we have this condition?

Reply:
The idea was to make sure we check the remote state only when needed and not for all cases. LocalSeq <=0 is the primary case. I wanted to also check when the active replicas is less that 50% of the total replicas. But after looking at the numbers, this may not be worth it